http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamException.scala

```diff
@@ -0,0 +1,29 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao.sam
+
+import org.broadinstitute.dsde.workbench.client.sam.ApiException
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoException
+import org.broadinstitute.dsde.workbench.model.TraceId
+
+/**
+ * Represents an exception interacting with Sam.
+ */
+class SamException private (message: String, code: Int, cause: Throwable, traceId: TraceId)
+    extends LeoException(
+      message,
+      code,
+      cause,
+      traceId = Some(traceId)
+    )
+
+object SamException {
+  def create(messagePrefix: String, apiException: ApiException, traceId: TraceId): SamException =
+    // TODO: I tried to use TCL SamExceptionFactory to process the ApiException.
```

Comment:
I just updated TCL to a later version in leo (like I merged it this morning) so it might be available now?

Reply:
I'll try again - the issue though was that `SamExceptionFactory` relies on Spring classes which Leo excludes. I'd rather not have Leo pull in Spring -- so maybe I'll take another look at the logic and resolve this TODO. 

(Ideally IMO a shared library like TCL shouldn't pull in Spring)

Reply:
Ah gotcha, yeah it is quite the pain to depend on TCL, I would rather not have leo pull in spring either, it is not worth the headache

Reply:
Looks like the [TCL code](https://github.com/DataBiosphere/terra-common-lib/blob/develop/src/main/java/bio/terra/common/sam/exception/SamExceptionFactory.java) just has some logic to extract a useful message from the generated client `ApiException` object. I'm going to do a little testing before blindly copying it to Leo, though :)

Reply:
Turns out the TCL logic was needed -- otherwise it includes the entire Sam response body, headers, etc in the exception message 😱 . Now it displays the nice message from Sam.